pkg/sessionctx/variable/session.go

```diff
@@ -1618,6 +1618,40 @@ type SessionVars struct {
 
 	// GroupConcatMaxLen represents the maximum length of the result of GROUP_CONCAT.
 	GroupConcatMaxLen uint64
+
+	// TablesBufferPool is a memory pool for table related memory allocation that aims to reuse memory
+	// and saves allocation
+	// The buffers are supposed to be used inside AddRecord/UpdateRecord/RemoveRecord.
+	// It's users duty to reset them before use.
+	TablesBufferPool TablesBufferPool
+}
+
+// TablesBufferPool is a memory pool for table related memory allocation that aims to reuse memory
+// and saves allocation
+type TablesBufferPool struct {
+	Update *UpdatePool
+	Add    *AddPool
+	Remove *RemovePool
```

Comment:
Just a question, why use pointer types here? Besides, these pools share some common fields (`[]int64`, `[]Datum`, `[]ColSize`), is it possible/safe to use a single union pool?

Reply:
> Just a question, why use pointer types here?

We use pointer types because we need to modify the fields in each pool. Without pointers, we would have to manually set the fields after usage, which would be less efficient and more error-prone.

> Besides, these pools share some common fields ([]int64, []Datum, []ColSize), is it possible/safe to use a single union pool?

Theoretically possible I think. But I don't see obvious benefits. The common slices are more of a coincidence rather than an intended design. Therefore, maintaining separate pools ensures clarity and avoids potential complications.